independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/update/rewrite/QuarkusUpdatesRepository.java

```diff
@@ -164,7 +174,13 @@ static List<String> fetchRecipesAsList(ResourceLoader resourceLoader, String loc
                                     }
                                     return null;
 
-                                }).filter(Objects::nonNull).collect(Collectors.toList());
+                                }).filter(Objects::nonNull).collect(Collectors.toMap(
+                                        sa -> sa[0],
+                                        sa -> sa[1],
+                                        (v1, v2) -> {
+                                            throw new IllegalStateException(String.format("Recipe with the same path detected. Can not happen! (%s)", v1));
+                                        },
```

Comment:
Should we merge them instead and ignore the dependency of the additional ones? Or we are sure this will never ever happen?

Reply:
@ia3andy WDYT?  From my POV the file list is "created" during the browsing of the filesystem. Key is the path (not only the file name) so there should be no duplicates.

Reply:
In fact there is no need to use a Map. Some kind of the List of the Pairs would be enough. I originally used List<Object[]>, but map looked better in my eyes. If needed I can refactor the code back to List of some Pair or Object[]....

Reply:
call the method `getRecipesByName`:)

Reply:
After consultation with @ia3andy, I changed the return type to the List<string[]>.